### DISCUSSION TITLE
Where are the log files?

### DISCUSSION THREAD

**Discussion by kosta** [getredash/redash#7130] [Q&A] [answered]
We've set up a machine according to the [setup guide](https://redash.io/help/open-source/setup/) and it's running fine.

But where are the log files for redash? I cant find them in systemd/journalctl, nor in `/var/log`. 

I found you can list the container using `ctr --namespace moby containers list --quiet` but it's kinda useless because they all look the same so using `for id in $(ctr --namespace moby containers list --quiet); do echo $id; ctr --namespace moby containers info $id | jq .Spec.process.args; echo; done;` as a workaround, I executed into the server container `ctr --namespace moby tasks exec -t --exec-id kosta $servercontainerid /bin/sh` but there's no systemd running there (no surprise) but also nothing in `/var/log`.

So where are the logs. Redash is writing logs, right?

**Comment by justinclift** [comment]
I'm guessing you're after the process logs?

If that's the case, your starting point is to look at the docker logs, as the containers are managed by docker.

So this should spit out everything:

```
$ docker compose logs
```

That specific command is generally not super useful without suitable options though.  For example telling it a specific container to get logs for, and potentially telling it to "follow" the logs (like with `tail`).  As an example:

```
$ docker compose logs -n 20 -f worker
```

That'll show the latest 20 lines of logs from the `worker` container, and the `-f` will cause the command to keep running and display new log lines as they occur.

Is this the kind of thing you're after? :smile:

**Reply by kosta** [reply]
Yes that the kind of thing I'm after but I don't think the containers are run via docker if you follow the setup guide (maybe from the developers guide? But this is a production machine).

In the VM that was created exactly via the guide, docker compose is not installed:

```
root@redash# docker compose logs
docker: 'compose' is not a docker command.
See 'docker --help'
```

Inside the container, run via containerd, docker is of course not installed either:

```
root@redash# ctr --namespace moby tasks exec -t --exec-id kosta 4f143479651bd70fc3b53ad29a92b2e63f53e28b31a06f5c3c8865f3ef74d57f /bin/sh
$ docker compose logs
/bin/sh: 1: docker: not found
$ 
```

So my question is: how can I see the logs of the machine I created via the [setup guide](https://redash.io/help/open-source/setup/), where the containers are run via containerd?

**Reply by justinclift** [reply, marked as answer]
Ahhh.  It sounds like we should revisit that page and update it.

We mostly point people to use the setup.sh script here, which sets things up in a "production" environment way: https://github.com/getredash/setup/

**Reply by kosta** [reply]
Ah so for prod, we should follow https://github.com/getredash/setup/ instead?

**Reply by justinclift** [reply]
Yep, definitely. :smile:

**Comment by jonahgreenthal** [comment]
I don't know if this is helpful or relevant, but it might be: When I was setting things up, one of the major obstacles I ran into is that `docker compose` (no hyphen) and `docker-compose` (hyphen) are different. Sometimes one works and the other doesn't, and I frequently had to adapt the instructions to change from one version to another. I think this has something to do with versions, but I don't understand the details.

**Reply by justinclift** [reply]
Yeah.  The original docker command didn't have a way to combine multiple containers into a single group.  So a third party invented the `docker-compose` approach.

As that worked really well, the `docker-compose` stuff was then incorporated into docker itself, and because `docker compose` (with the space).

So, `docker-compose` is an early approach (docker "v1" kind of thing), whereas `docker compose` is the newer approach.  With some care, both cane be installed on the same system at the same time.  That's only useful for specific circumstances though. :wink:
